SU-HSDO/suhumsci
# Fix(STN-892) : Mobile nav link expand icon placement

# READY FOR REVIEW

## Summary
When a site has a search box in the nav while a nav item has nested items and is also a link there is a positioning issue with the +/- icon where it is not being contained within it’s corresponding nav item. This is happening for top level and nested nav items. 

This work just adds a little more specificity to a style that was being added so the Main Menu Nav button is in a correct spot whether there is a search bar or not a search bar added to the site.  The non-search bar sites seem to be the minority, but it's good to have it accounted for.

I added a class instead of "button" as that was too broad and was hitting every "+" and "-" button in the menu if there was a search bar present and moving it the position way too high.

https://sparkbox.atlassian.net/browse/STN-892

## Need Review By (Date)
10/29

## Urgency
medium

## Steps to Test
1. npm run test
2. First lower your page down to a tablet or mobile size to see the mobile menu, double-check all of the "+" and "-" buttons work correctly.
3. Add a search bar to the nav Search region area: http://sparkbox-sandbox.suhumsci.loc/admin/structure/block (Structure > Block Layout), scroll down to the "Search" region and hit "Place Block" from there you will place the block: "Exposed form: hs_default_search-search_page" and be sure to uncheck "Show title" before adding and saving the block.
4. Once you have a search block loading in the Nav area, once again check your mobile menu and make sure the icons stay in the correct spots for each link.

## Verifications
- [x] Verify Steps to Test Step 2 in the Steps to Test. Without a search bar, the menu link icons should be in the same spot.
- [x] Verify Steps to Test Step 4 in the Steps to Test. With a search bar, the menu link icons should be in the same spot.
- [x] Verify the icons are correct when a nested link is at the top of the menu.
- [x] Verify the icons are correct when an unpublished page nested link is the top link or in the menu.
- [x] Verify the icons are correct when a private page nested link is the top link or in the menu.
- [x] Verify the Main Menu Toggle is in the correct position, both with a search bar and without.
- [x] Verify the Main Menu Toggle and the menu link icons are correct on both themes, Traditional and Colorful. (you will have to place the search block for each theme)

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
